### PR TITLE
Send a proper response for status.php on trusted domain error

### LINKED ISSUE #7732: "Access through untrusted domain"
When having not properly setup "trusted_domains" we get on nextcloud android app only "unknown error occured".
We are calling "server/status.php" and get 400 back.
Is 400 only returned in this case, so it is safe to check for this? Or is there a more decent way to get this?

Ratio: from time to time home users approach to us and have it misconfigured. So I want to improve this a bit

### PR DESCRIPTION
**Pull request by MorrisJobke**
* fixes #7732

Maybe good to backport to stable12 and stable13 to make it easier for apps to detect the real problem.

### CODE REVIEW

File: lib/base.php [hunk 1]
```diff
@@ -779,8 +779,16 @@ public static function init() {
 				$isScssRequest = true;
 			}
 
+			if(substr($request->getRequestUri(), -11) === '/status.php') {
+				OC_Response::setStatus(\OC_Response::STATUS_BAD_REQUEST);
+				header('Status: 400 Bad Request');
+				header('Content-Type: application/json');
+				echo '{"error": "Trusted domain error.", "code": 15}';
```
Comment (MorrisJobke) [inline lib/base.php]: I have chosen 15 for no particular reason.

Comment (rullzer) [reply lib/base.php]: When in doubt always use 42!

File: lib/base.php [hunk 2]
```diff
@@ -779,8 +779,16 @@ public static function init() {
...
+				exit();
+			}
+
 			if (!$isScssRequest) {
-				header('HTTP/1.1 400 Bad Request');
+				OC_Response::setStatus(\OC_Response::STATUS_BAD_REQUEST);
```
Comment (rullzer) [inline lib/base.php]: Why change this?

**Review by rullzer — APPROVED** [review]
Fine by me

File: lib/base.php (hunk 2 above)
Comment (MorrisJobke) [reply lib/base.php]: Because I was in that code and this makes then use of the proper HTTP/1.1 vs HTTP/2.0 split ;)

Comment (rullzer) [reply lib/base.php]: okidoki!

**Comment by tobiasKaminsky** [comment]
I already implemented in on android. Works great 👍

**Comment by tobiasKaminsky** [comment]
Do you guys have an idea what I can display instead of simply "Untrusted domain."?
Some more useful infos/ a link?

**Comment by rullzer** [comment]
@tobiasKaminsky maybe something like:

'It seems your Nextcloud is not properly setup on this domain. Please go to <url> in a browser for more information.'

**Comment by tobiasKaminsky** [comment]
Thanks @rullzer.
I already came up with "Access through untrusted domain. Please see documentation for further infos." as this has the right keywords in it.
Hope it is fine ;-)

File: lib/base.php (hunk 1 above)
Comment (juliushaertl) [reply lib/base.php]: We should have this documented somewhere. Do we have any existing documentation about status.php?

Comment (rullzer) [reply lib/base.php]: I don't think so. But yes you are right.

**Review by tobiasKaminsky — APPROVED** [review]
Already implemented on android & working
👍

**Pull request merged**